src/geometry/manifold/instances/sphere.lean

```diff
@@ -285,3 +295,92 @@ instance {n : ℕ} [_i : fact (findim ℝ E = n + 1)] :
   chart_mem_atlas  := λ v, ⟨-v, rfl⟩ }
 
 end charted_space
+
+section smooth_manifold
+
+variables [finite_dimensional ℝ E]
```

Comment:
I have the impression this one is not necessary for all the statements below: if you have `findim E = n + 1`, then in particular `findim E` is not zero, and therefore `E` is finite-dimensional.

Reply:
I've addressed this by adding
```lean
lemma finite_dimensional_of_findim_eq_succ {K V : Type*} [field K] [add_comm_group V]
  [vector_space K V] (n : ℕ) [fact (findim K V = n + 1)] :
  finite_dimensional K V :=
```
in the file `linear_algebra.finite_dimensional`, and invoking it as a local instance where needed.  Is this an elegant solution or a hack?  You decide :)